cookbooks/oceanic_extension/oceanic_extension.prm

```diff
@@ -0,0 +1,230 @@
+# The first represents strain accrued during brittle (plastic) deformation. 
+# The initial values of this field are used to localize deformation in the center of the model away from the boundaries. 
+# The remaining two compositional fields represent distinct lithologies for the oceanic crust and mantle lithosphere.
+subsection Compositional fields
+  set Number of fields = 3
+  set Names of fields = plastic_strain, oceanic_crust, mantle_lithosphere
+end
+
+
+# Spatial domain of different compositional fields
+# JBN - expand documenation here
+subsection Initial composition model
+  set Model name = function
+  subsection Function
+  # The function rand_seed(seed_number) will also produce the same random value at each point when the parameter file is run.
+  # "seed_number" is an integer and changing this value will produce a different (reproducible) random number. 
+  # A distinct random number can be produced each time the simulation is run by replacing rand_seed(seed_number) with rand().
+    set Variable names      = x,y
+    set Function expression = if(x>150.e3 && x<250.e3, 0.5 + rand_seed(0)*1.0, 0);  \
+			                        if(y>=93.e3, 1, 0); \
+                              if(y<93.e3 && y>-100.e3, 1, 0); 
+  end
+end
+
+
+# Temperature boundary conditions
+# Prescriibed temperature at bottom and top boundaries with values from the initial temperature field
+subsection Boundary temperature model
+  set Fixed temperature boundary indicators = bottom, top
+  set Model name = initial temperature
+end
+
+# Initial temperature field
+# JBN - expand documentation here
+subsection Initial temperature model
+  set List of model names = function
+  subsection Function
+    set Variable names = x,y
+    set Function constants = ymax=100e3, ageop=1.26e15, Tm=1573, Ts=273, kappa=1.0101e-6
+    set Function expression = (Ts + (Tm-Ts)*(1-erfc((ymax-y)/(2*sqrt(kappa*ageop)))))
+  end
+end
+
+
+# Material model
+subsection Material model
+  set Model name = visco plastic
+
+  # JBN - add comment here about what this parameter does and why we choose "harmonic average" (or a different value)
+  set Material averaging = harmonic average
+
+  subsection Visco Plastic
+
+    set Reference temperature = 273
+    set Reference viscosity   = 1e21
+    set Minimum strain rate = 1.e-20
+    set Reference strain rate = 1.e-15
+    # AG - maybe comment on how to choose the right min/max
+    set Minimum viscosity = 1e18
+    set Maximum viscosity = 1e26
+
+    set Define thermal conductivities = true
+    set Thermal conductivities = 2.5
+    set Densities             =        3300,        3300,        2900,        3300
+    set Heat capacities       = 750.
+    set Thermal expansivities = 2e-5
+
+    # JBN - Add comment explaining what this value does and how it affects solution
+    set Viscosity averaging scheme = maximum composition
+
+    set Viscous flow law = composite
+
+    # JBN - add documenation about where these values come from
```

Comment:
Replace with references for these flow laws? Can you also add a note that the second column of values represent the plastic strain field and are not actually used in the calculation. One strategy I use is to set the values in this column to clearly distinct/incorrect values, such as 1e-50, 1.0, 0, 0. The last four values would produce a viscosity of 5e49 Pa s.

Reply:
Shall I set incorrect values also for the dislocation creep values in the second column then?